MbDotNet/Models/Predicates/Fields/PredicateGeneratorFields.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using MbDotNet.Enums;
+using Newtonsoft.Json;
+
+namespace MbDotNet.Models.Predicates.Fields
+{
+    public class PredicateGeneratorFields : PredicateFields
```

Comment:
I don't think this type is going to be enough. It is specific to the predicate fields of an HTTP imposter predicate, but we'll need to support TCP as well (see `TcpPredicateFields`).

I'm not very familiar with proxies in mountebank, but the documentation appears to show that you can use the regular predicate fields with values like this:
```
"predicateGenerators": [{
    "matches": {
        "query": { "q": "mountebank" }
    }
}]
```

Or the boolean version that you are supporting here:
```
"predicateGenerators": [{
    "matches": { "query": true },
    "caseSensitive": true
}]
```

I think we need to find a way to have HTTP and TCP value predicate fields and HTTP and TCP boolean predicate fields. It's something I've needed in the past in order to implement things like the [exists predicate](http://www.mbtest.org/docs/api/predicates#predicates-exists), but never came up with a good way of doing it. One concern is that I don't think we would want people to be creating the other predicate types, like `EqualsPredicate` with the boolean version of the fields since it doesn't really make sense.

I'll give this some more thought, let me know if you have any ideas on how we can represent both the value-style and boolean-style predicate fields while still restricting which types of predicates can use which predicate field types.

Reply:
I guess worst-case scenario we just define the two styles of predicate fields and don't restrict them, and if someone uses boolean fields on an equals predicate or something it just won't work. I'd rather restrict it somehow though so people can't create invalid predicates.

Reply:
Also, I guess you don't need to add the implementations for TCP as long as the final design is able to support adding it. I can always add it later.